assets/component-slideshow.css

```diff
@@ -0,0 +1,271 @@
+slideshow-component {
+  position: relative;
+  display: block;
+}
+
+.spaced-section--full-width:first-child slideshow-component.page-width {
+  margin-top: 5rem;
+}
+
+.spaced-section--full-width:last-child slideshow-component.page-width {
+  margin-bottom: 5rem;
+}
+
+@media screen and (min-width: 750px) {
+  .spaced-section--full-width:first-child slideshow-component.page-width {
+    margin-top: calc(5rem + var(--page-width-margin));
+  }
+
+  .spaced-section--full-width:last-child slideshow-component.page-width {
+    margin-bottom: calc(5rem + var(--page-width-margin));
+  }
+}
+
+@media screen and (max-width: 989px) {
+  .no-js slideshow-component .slider {
+    padding-bottom: 3rem;
+  }
+}
+
+.slideshow.banner {
+  flex-direction: row;
+  flex-wrap: nowrap;
+  margin: 0;
+}
+
+.slideshow__slide {
+  padding: 0;
+  position: relative;
+  display: flex;
+  flex-direction: column;
+}
+
+@media screen and (max-width: 749px) {
+  .slideshow--placeholder.banner--mobile-bottom.banner--adapt_image .slideshow__media,
+  .slideshow--placeholder.banner--adapt_image:not(.banner--mobile-bottom) {
+    height: 28rem;
+  }
+}
+
+@media screen and (min-width: 750px) {
+  .slideshow--placeholder.banner--adapt_image {
+    height: 56rem;
+  }
+}
+
+.slideshow__text.banner__box {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  max-width: 54.5rem;
+}
+
+@media screen and (max-width: 749px) {
+  .banner--mobile-bottom .slideshow__text.banner__box {
+    max-width: 100%;
+  }
+
+  .banner--mobile-bottom .slideshow__text-wrapper {
+    flex-grow: 1;
+  }
+
+  .banner--mobile-bottom .slideshow__text.banner__box {
+    height: 100%;
+  }
+
+  .banner--mobile-bottom .slideshow__text .button {
+    flex-grow: 0;
+  }
+
+}
+
+.banner:not(.banner--mobile-bottom) .slideshow__text-wrapper {
+  height: 100%;
+}
+
+@media screen and (min-width: 750px) {
+  .slideshow__text-wrapper {
+    height: 100%;
+    padding: 5rem;
+  }
+
+  .slideshow__text-wrapper.slideshow__text-wrapper--top-left {
+    align-items: flex-start;
+    justify-content: flex-start;
+  }
```

Comment:
You're right, I removed the `.class.class` and we're still good. 

As for the classes, the way it is right now is easy in the sense that I just have to use the setting value at the end of the class. I find it clearer than having to manipulate the value of the setting by doing some liquid string manipulation to get the values from it. 

For `top-left` I'd need to add two classes for `top` and for `left`.

Reply:
I think for that, you'd just have the value of the setting be `slideshow__text-wrapper--top  slideshow__text-wrapper--left`.  Then you don't need to do any string manipulation.

Reply:
Tried this out ^ but that ends up being too many characters. Tried with shorter class name but was still having an error locally. 

One way I could go about it is with CSS selectors like this but it might not be as legible: 

```css
[class*="slideshow__text-wrapper--top"] {
    align-items: flex-start;
  }

.slideshow__text-wrapper[class*="-left"] {
    justify-content: flex-start;
  }
```